waiter/src/waiter/state/maintainer.clj

```diff
@@ -628,11 +628,19 @@
                                                                               (dissoc service-id->instability-issue service-id))]
                                          (when (or (not= (get service-id->healthy-instances service-id) healthy-instances)
                                                    (not= (get service-id->unhealthy-instances service-id) unhealthy-instances))
-                                           (let [curr-instance-ids (set (map :id healthy-instances))
-                                                 prev-instance-ids (set (map :id (get service-id->healthy-instances service-id)))
-                                                 new-instance-ids (filterv (complement prev-instance-ids) curr-instance-ids)
-                                                 rem-instance-ids (filterv (complement curr-instance-ids) prev-instance-ids)
+                                           (let [curr-instances (set healthy-instances)
+                                                 prev-instances (set (get service-id->healthy-instances service-id))
+                                                 new-instances (filterv (complement prev-instances) curr-instances)
+                                                 rem-instances (filterv (complement curr-instances) prev-instances)
```

Comment:
These can remain as `filter` instead of `filterv`

Reply:
Done